### PR TITLE
New registration option for agent that already got a key

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#9569|

## Description

If the agent already has a key, it will send it to the manager for validation.

## Dod

- Compilation without warnings in every supported platform
  - [X] Linux
- [X] Source installation
- [X] Review logs syntax and correct language

### CODE REVIEW

File: src/shared/enrollment_op.c [hunk 1]
```diff
@@ -477,6 +485,25 @@ static void w_enrollment_verify_ca_certificate(const SSL *ssl, const char *ca_ce
     }
 }
 
+/**
+ * @brief Concats the current key of the agent, if exist, as  part of the enrollment message
```
Comment (DProvinciani) [inline src/shared/enrollment_op.c:489]: `Concatenates` the current......

File: src/shared/enrollment_op.c [hunk 2]
```diff
@@ -477,6 +485,25 @@ static void w_enrollment_verify_ca_certificate(const SSL *ssl, const char *ca_ce
...
+ *
+ * @param buff buffer where the KEY section will be concatenated
+ * @param key key will be added
```
[the remark this answers is not in the document]
Comment (DProvinciani) [reply src/shared/enrollment_op.c:492]: I suggest rewording the description of the parameter as `The key that will be concatenated`.

File: src/shared/enrollment_op.c [hunk 3]
```diff
@@ -477,6 +485,25 @@ static void w_enrollment_verify_ca_certificate(const SSL *ssl, const char *ca_ce
...
+ */
+static void w_enrollment_concat_key(char *buff, keyentry* key_entry) {
+    assert(buff != NULL);
+    assert(key_entry != NULL);
+
+    os_sha1 output;
+    char* opt_buf = NULL;
+    os_calloc(OS_SIZE_65536, sizeof(char), opt_buf);
```
Comment (DProvinciani) [inline src/shared/enrollment_op.c:500]: I suggest using a smaller size of memory. Probably `OS_SIZE_512` or `OS_SIZE_1024` would be enough.

**Review by DProvinciani — CHANGES_REQUESTED** [review]
Just some suggestions. Apply changes according to the comments.

File: src/shared/enrollment_op.c (hunk 3 above)
Comment (palaciosjeremias) [reply src/shared/enrollment_op.c:500]: As we know the exact length of os_sha1 with sizeof(os_sha1) we can calculate exactly how much memory it will cost.
For code clarity, we can use OS_SIZE_512 as Diego suggested,

File: src/shared/enrollment_op.c (hunk 1 above)
Comment (MiguelazoDS) [reply src/shared/enrollment_op.c]: Done

File: src/shared/enrollment_op.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply src/shared/enrollment_op.c]: Done

File: src/shared/enrollment_op.c (hunk 3 above)
Comment (MiguelazoDS) [reply src/shared/enrollment_op.c]: Fixed

Comment (pereyra-m) [reply src/shared/enrollment_op.c]: I also think that the variable size is much bigger than necessary, but there are also many places around this code where the memory allocated is huge and we could review them later.

**Review by pereyra-m — APPROVED** [review]
LGTM!

**Review by DProvinciani — APPROVED** [review]
Looks good!
